Core.ApplicationServices/Contract/ItContractService.cs

```diff
@@ -286,6 +330,30 @@ public Result<IEnumerable<(int year, int quarter)>, OperationError> GetAppliedPr
                 );
         }
 
+        public Maybe<OperationError> SetContractResponsibleUnit(int contractId, Guid targetUnitUuid)
```

Comment:
service method not covered by unit tests (the domain is so what you need to test here is that the prerequisites for calling the domain are true and that successful result from the domain result in the side effects you expect for both success and error cases...
for success you expect domain events and save + transaction
for error you expect rollback and no domain events and no save etc...

your domain test should cover the different branches inside the domain call.